corehq/apps/hqwebapp/views.py

```diff
@@ -1368,3 +1382,29 @@ def post(self, request, *args, **kwargs):
             ))
 
         return HttpResponseRedirect(reverse('oauth2_provider:detail', args=[str(base_application.id)]))
+
+
+@require_POST
+def check_sso_login_status(request):
+    """
+    Checks to see if a given username must sign in or sign up with SSO and
+    returns the url for the SSO's login endpoint.
+    :param request: HttpRequest
+    :return: HttpResponse (as JSON)
+    """
+    username = request.POST['username']
+    is_sso_required = False
+    sso_url = None
+    continue_text = None
+
+    idp = IdentityProvider.get_required_identity_provider(username)
+    if idp:
+        is_sso_required = True
+        sso_url = idp.get_login_url(username=username)
+        continue_text = _("Continue to {}").format(idp.name)
```

Comment:
Should this be evaluated if `ENFORCE_SSO_LOGIN` is false?

Reply:
no, as this has nothing to do with forcing someone to login or not—that is all handled by the sign up or sign in views themselves. This is just an informational/utility view and should accurately report the status.